posthog/models/insight.py

```diff
@@ -124,7 +124,11 @@ def query_from_filters(self):
         from posthog.hogql_queries.legacy_compatibility.filter_to_query import filter_to_query
 
         try:
-            return {"kind": "InsightVizNode", "source": filter_to_query(self.filters).model_dump(), "full": True}
+            return {
+                "kind": "InsightVizNode",
+                "source": filter_to_query(self.filters).model_dump(exclude_unset=True, exclude_none=True),
```

Comment:
I've noticed that the `exclude_unset` also removes the `kind` property and other set-by-default properties from the source. For example, this query:
```python
{'full': True, 'kind': 'DataTableNode', 'source': {'kind': 'ActorsQuery', 'properties': [{'key': 'id', 'type': 'cohort', 'value': 4669}]}}

```
Becomes:
```python
{'full': True, 'kind': 'DataTableNode', 'source': {'properties': [{'key': 'id', 'value': 4669}]}}
```
I think we want to keep those properties.

Reply:
Ah yeah definitely. Don't know why I commited with this setting. Good catch!